### PR TITLE
add gene limit:100

### PR DESCRIPTION
**Pull request by dippindots**
Do not allow queries with more than 100 genes
Fix https://github.com/cBioPortal/cbioportal/issues/5932.

### CODE REVIEW

**Comment by dippindots** [comment]
We can use this [link](https://cbioportal-frontend-pr-2194.herokuapp.com/results/oncoprint?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_list=cellline_ccle_broad&case_set_id=cellline_ccle_broad_cnaseq&data_priority=0&gene_list=ADAM10%2520ADAM17%2520APH1A%2520APH1B%2520ARRDC1%2520CIR1%2520CTBP1%2520CTBP2%2520CUL1%2520DLL1%2520DLL3%2520DLL4%2520DTX1%2520DTX2%2520DTX3%2520DTX3L%2520DTX4%2520EP300%2520FBXW7%2520HDAC1%2520HDAC2%2520HES1%2520HES5%2520HEYL%2520ITCH%2520JAG1%2520JAG2%2520KDM5A%2520LFNG%2520MAML1%2520MAML2%2520MAML3%2520MFNG%2520NCOR2%2520NCSTN%2520NOTCH1%2520NOTCH2%2520NOTCH3%2520NOTCH4%2520NRARP%2520NUMB%2520NUMBL%2520PSEN1%2520PSEN2%2520PSENEN%2520RBPJ%2520RBPJL%2520RFNG%2520SNW1%2520SPEN%2520HES2%2520HES4%2520HES7%2520HEY1%2520HEY2%2520RB1%2520RBL1%2520RBL2%2520CCNA1%2520CCNB1%2520CDK1%2520CCNE1%2520CDK2%2520CDC25A%2520CCND1%2520CDK4%2520CDK6%2520CCND2%2520CDKN2A%2520CDKN2B%2520MYC%2520CDKN1A%2520CDKN1B%2520E2F1%2520E2F2%2520E2F3%2520E2F4%2520E2F5%2520E2F6%2520E2F7%2520E2F8%2520SRC%2520JAK1%2520JAK2%2520STAT1%2520STAT2%2520STAT3%2520STAT5A%2520STAT5B%2520AKR1C3%2520AR%2520CYB5A%2520CYP11A1%2520CYP11B1%2520CYP11B2%2520CYP17A1%2520CYP19A1%2520CYP21A2%2520HSD17B1%2520HSD17B10%2520HSD17B11%2520HSD17B12%2520HSD17B13%2520HSD17B14%2520HSD17B2%2520HSD17B3%2520HSD17B4%2520HSD17B6%2520HSD17B7%2520HSD17B8%2520HSD3B1%2520HSD3B2%2520HSD3B7%2520RDH5%2520SHBG%2520SRD5A1%2520SRD5A2%2520SRD5A3%2520STAR&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=cellline_ccle_broad_cna&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=cellline_ccle_broad_mutations&tab_index=tab_visualize) to test the result page with more than 100 genes.

File: src/config/IAppConfig.ts [hunk 1]
```diff
@@ -107,5 +107,6 @@ export interface IServerConfig {
     "sessionServiceEnabled": boolean;
     "session_url_length_threshold":string;
     "mskWholeSlideViewerToken":string;
+    "gene_limit": number;
```
Comment (alisman) [inline src/config/IAppConfig.ts]: lets called "query_gene_limit"

Comment (dippindots) [reply src/config/IAppConfig.ts]: sure

File: src/shared/components/query/QueryStore.ts [hunk 2]
```diff
@@ -1457,6 +1457,9 @@ export class QueryStore {
             }
         }
 
+        if (this.oql.query.length > AppConfig.serverConfig.gene_limit) {
+            return "Gene limit is 100, please keep the number of genes in range.";
```
Comment (alisman) [inline src/shared/components/query/QueryStore.ts]: lets just say: "Please limit your queries to 100 genes or fewer."

File: src/pages/resultsView/ResultsViewPage.tsx [hunk 3]
```diff
@@ -411,7 +412,19 @@ export default class ResultsViewPage extends React.Component<IResultsViewPagePro
         }
     }
 
+    @computed get isPageValid() {
```
Comment (alisman) [inline src/pages/resultsView/ResultsViewPage.tsx]: Nice!, but lets call it "isQueryValid"

Comment (dippindots) [reply src/pages/resultsView/ResultsViewPage.tsx]: Thanks! Updated.

File: src/shared/components/query/GeneSymbolValidator.tsx [hunk 4]
```diff
@@ -39,6 +39,16 @@ export default class GeneSymbolValidator extends QueryStoreComponent<{}, {}>
 		if (!this.store.oql.query.length)
 			return null;
 
+		if (this.store.isQueryLimitReached)
+			return (
+				<div className={styles.GeneSymbolValidator}>
+					<div className={styles.invalidBubble} title="Please limit your queries to 100 genes or fewer.">
+						<FontAwesome className={styles.icon} name='exclamation-circle'/>
+						<span>Please query no more than 100 genes.</span>
```
Comment (alisman) [inline src/shared/components/query/GeneSymbolValidator.tsx]: @dippindots lets add the language about contact us if use case requires more ...

Comment (alisman) [reply src/shared/components/query/GeneSymbolValidator.tsx]: lets just make it read exactly the same: 
"Queries are limited to 100 genes. Please <a href={`mailto:${AppConfig.serverConfig.skin_email_contact}`}>let us know</a> your use case(s) if you need to query more than 100 genes."

**Comment by alisman** [comment]
@jjgao there are 4 failures:
3 timeouts and patient page screenshot failure which is data related
going to merge this

**Pull request merged**